scripts/join-metadata-and-clades.py

```diff
@@ -82,7 +82,24 @@ def main():
     args = parse_args()
 
     metadata = pd.read_csv(args.first_file, index_col=METADATA_JOIN_COLUMN_NAME,
-                           sep='\t', low_memory=False, na_filter = False)
+                           sep='\t', low_memory=False)
+
+    # Check for existing annotations in the given metadata. Skip join with
+    # Nextclade QC file, if those annotations already exist and none of the
+    # columns have empty values. In the case where metadata were combined from
+    # different sources with and without annotations, the "clock_deviation"
+    # column will exist but some values will be missing. We handle this case as
+    # if the annotations do not exist at all and reannotate all columns. We
+    # cannot look for missing values across all expected columns as evidence of
+    # incomplete annotations, since a complete annotation by Nextclade will
+    # include missing values for some columns by design.
```

Comment:
I have ~no context on the columns produced by Nextclade, but in general reannotating all rows when there is a mix of annotated and unannotated rows seems good to me, as it means the resulting annotations will be consistently from a single Nextclade version. (OTOH, it probably adds a bit of not-strictly-necessary runtime.)

My thought had been to handle mixed annotated/unannotated rows by (1) always annotating everything and then (2) looking at the `*_original` columns to copy back/preserve the existing annotations in preference to the new annotations. But this solution of yours seems better!